### PR TITLE
KV async job mgr

### PR DESCRIPTION
**Pull request by bluesalt**
This PR implements a new async pending job management design which persists the jobs into RocksDB to reduce space usage and improve performance.

### CODE REVIEW

**Comment by bluesalt** [comment]
@peteryj

File: cmd/auklet/command/dump_db.go [hunk 1]
```diff
@@ -66,12 +69,17 @@ func (c *DumpDBCommand) Run(args []string) int {
 	pre := []byte(prefix)
 	for iter.Seek(pre); iter.ValidForPrefix(pre); iter.Next() {
 		b := iter.Value().Data()
-		index := new(pack.DBIndex)
-		if err := proto.Unmarshal(b, index); err != nil {
+		var obj proto.Message
+		if keyType == "index" {
+			obj = new(pack.DBIndex)
+		} else {
```
Comment (abitduck) [inline cmd/auklet/command/dump_db.go]: keyType可以指定一下，改为bool型也行，可降低用户使用门槛

File: common/fs/mount_monitor.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package fs
```
Comment (abitduck) [inline common/fs/mount_monitor.go]: 这个文件没有Copyright信息，也没看到引用

File: common/fs/utils.go [hunk 3]
```diff
@@ -114,7 +114,7 @@ func GetFileMTime(filePath string) (int64, error) {
 	return info.ModTime().Unix(), nil
 }
 
-func TempDir(driveRoot string, device string) string {
+func TempDir(driveRoot, device string, policy int) string {
```
Comment (abitduck) [inline common/fs/utils.go]: var policy is unused, should have a comment if needed

File: objectserver/errors.go [hunk 4]
```diff
@@ -20,12 +20,17 @@ import (
...
-	PathNotDirError     = errors.New("path is not a directory")
-	NotPackEngine       = errors.New("engine is not pack type")
-	DriveFull           = errors.New("drive is full")
-	ErrHashConfNotFound = errors.New("unable to read hash prefix and suffxi")
+	EngineNotFound          = errors.New("object engine not found")
+	LockPathError           = errors.New("unable to lock path")
+	PathNotDirError         = errors.New("path is not a directory")
+	NotPackEngine           = errors.New("engine is not pack type")
+	DriveFull               = errors.New("drive is full")
+	ErrHashConfNotFound     = errors.New("unable to read hash prefix and suffxi")
```
Comment (abitduck) [inline objectserver/errors.go]: spell error: `suffxi` to `suffix`

File: objectserver/fs_async_job_mgr.go [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+			if cnt > int64(BLOOMFILTER_RESET_THREASHHOLD) {
+				glogger.Info("reset bloom filter", zap.Int64("elements", cnt))
+				s.filter.Clear()
+			}
+
+			glogger.Debug("add unlisted entry",
+				zap.String("entry", j), zap.Int64("elements", cnt))
+
+			b, err := ioutil.ReadFile(filepath.Join(d, j))
+			if err != nil {
```
Comment (abitduck) [inline objectserver/fs_async_job_mgr.go:164]: if read file fail too many times, should raise exception, disk error or file permission may cause this problem; always continue is not a good idea~

File: cmd/auklet/command/dump_db.go (hunk 1 above)
Comment (abitduck) [inline cmd/auklet/command/dump_db.go:80]: should give  keyType for KVAsync a name, and should update the command usage

File: common/fs/mount_monitor.go (hunk 2 above)
Comment (abitduck) [inline common/fs/mount_monitor.go:1]: This file do not have copyright info and code is unused

**Review by abitduck — CHANGES_REQUESTED** [review]
Great feature comes ~ some code need have a tweak

File: common/fs/utils.go (hunk 3 above)
Comment (bluesalt) [reply common/fs/utils.go]: It is a bug. Will fix it.

File: objectserver/fs_async_job_mgr.go (hunk 5 above)
Comment (bluesalt) [reply objectserver/fs_async_job_mgr.go:164]: That is why bloom filter is used.

**Review by abitduck — APPROVED** [review]

**Pull request merged**